### PR TITLE
fix: Stack Overflow plugin search highlight fix

### LINKED ISSUE #22531: 🐛 Bug Report: StackOverflowSearchResultListItem doesn't decode highlighted text & author names
### 📜 Description

`StackOverflowSearchResultListItem` from [Stack Overflow plugin](https://github.com/backstage/backstage/blob/master/plugins/stack-overflow/)  doesn't decode `title` when using the `highlight` prop which results in special characters, (e.g., `&#39;`) being displayed in their encoded form in the title, instead of the actual characters (e.g., `'`).

The same problem applies to `author` names, with and without `highlight` so if your name contains non-ASCII characters, these are not being decoded correctly. As a result, names with extended characters (such as `å, ä, ö`) are displayed in their encoded forms.

### 👍 Expected behavior

Content should render in their decoded form.

### 👎 Actual Behavior with Screenshots

1. Highlighted text
image

2. Author name with non-ASCII character
image

### 👟 Reproduction steps

1. Follow the steps on how to add the [StackOverflowSearchResultListItem](https://github.com/backstage/backstage/tree/master/plugins/stack-overflow#use-specific-search-result-list-item-for-stack-overflow-question).
2. Add `highlight={result.highlight}` to component
3. Search for questions with special characters

### 📃 Provide the context for the Bug.

_No response_

### 🖥️ Your Environment

_No response_

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by OSfrog**
## Hey, I just made a Pull Request!
This fixes the issue in `StackOverflowSearchResultListItem` where `authors` and highlighted `titles` weren't rendering the decoded values.

Decoding issues in title and author: 
image
image

Fix:
image
image

Closes #22531 
#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/stack-overflow/package.json [hunk 1]
```diff
@@ -54,6 +54,7 @@
     "@testing-library/jest-dom": "^6.0.0",
     "@types/react": "^16.13.1 || ^17.0.0",
     "cross-fetch": "^4.0.0",
+    "he": "^1.2.0",
```
Comment (freben) [inline plugins/stack-overflow/package.json:57]: Oof, 124 kB for this! That's heavy for a one-time use. Just noting.

You need to run `yarn install` and commit the `yarn.lock` changes along with the PR.

**Review by freben — COMMENTED** [review]
Ping @backstage/discoverability-maintainers

File: plugins/stack-overflow/package.json (hunk 1 above)
Comment (OSfrog) [reply plugins/stack-overflow/package.json:57]: Noted, I'll see if I can come up with a better solution!

File: plugins/stack-overflow/src/search/StackOverflowSearchResultListItem/StackOverflowSearchResultListItem.tsx [hunk 2]
```diff
@@ -48,6 +48,10 @@ export const StackOverflowSearchResultListItem = (
   const analytics = useAnalytics();
 
   const handleClick = () => {
+    if (!result) {
+      return;
```
Comment (djamaile) [inline plugins/stack-overflow/src/search/StackOverflowSearchResultListItem/StackOverflowSearchResultListItem.tsx:52]: shouldn't it return `null` like that if statement below?

**Review by djamaile — COMMENTED** [review]
Still some issues with CI, any chance you could have a look?

File: plugins/stack-overflow/src/search/StackOverflowSearchResultListItem/StackOverflowSearchResultListItem.tsx (hunk 2 above)
Comment (OSfrog) [reply plugins/stack-overflow/src/search/StackOverflowSearchResultListItem/StackOverflowSearchResultListItem.tsx:52]: Not sure why I added an extra check there, will remove.

**Comment by OSfrog** [comment]
@freben I've added another solution, please let me know if this is okay or if there is anything else I can do 😄

**Comment by OSfrog** [comment]
> Still some issues with CI, any chance you could have a look?

Should be resolved now

**Review by djamaile — APPROVED** [review]

**Comment by OSfrog** [comment]
ping @freben @djamaile

**Review by freben — COMMENTED** [review]
Thank you. Just double checking about that potential element leak

File: plugins/stack-overflow/src/util.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export function decodeHtml(input: string) {
+  const textContainer = document.createElement('textarea');
+  textContainer.innerHTML = input;
+  return textContainer.value;
```
[the remark this answers is not in the document]
Comment (OSfrog) [reply plugins/stack-overflow/src/util.ts:20]: Yes, this should be garbage collected

**Comment by freben** [comment]
remember the DCO signature on each commit!

**Comment by OSfrog** [comment]
Accidentally committed through the GUI haha, how can I fix this?

**Comment by OSfrog** [comment]
should be all good now

**Review by freben — APPROVED** [review]

**Pull request merged**